static/src/javascripts/projects/common/modules/atoms/youtube-player.spec.ts

```diff
@@ -0,0 +1,189 @@
+import type {
+	Callback,
+	ConsentState,
+} from '@guardian/consent-management-platform/dist/types';
+import { _ as youtubePlayer } from 'common/modules/atoms/youtube-player';
+
+jest.mock('common/modules/commercial/build-page-targeting', () => ({
+	getPageTargeting: jest.fn(() => ({ key: 'value' })),
+}));
+
+jest.mock('@guardian/commercial-core', () => ({
+	...jest.requireActual('@guardian/commercial-core'),
+	getPermutivePFPSegments: jest.fn(() => [42]),
+}));
+
+jest.mock('lib/config', () => ({
+	get: jest.fn((key: string, fallback?: unknown) => {
+		if (key === 'page.adUnit') {
+			return 'adunit';
+		}
+		if (key === 'isDotcomRendering') {
+			return false;
+		}
+
+		if (fallback) return fallback;
+
+		throw new Error(
+			`Unexpected config lookup '${key}', check the mock is still correct`,
+		);
+	}),
+}));
+
+jest.mock('@guardian/consent-management-platform', () => ({
+	onConsentChange: jest.fn((callback: Callback) =>
+		callback({
+			tcfv2: {
+				consents: { 1: true },
+				gdprApplies: true,
+				eventStatus: 'tcloaded',
+				vendorConsents: { abc: true },
+				tcString: 'testTcString',
+				addtlConsent: 'testaddtlConsent',
+			},
+		}),
+	),
+}));
+
+jest.mock('common/modules/commercial/commercial-features', () => ({
+	commercialFeatures: jest.fn(),
+}));
+
+jest.mock('common/modules/experiments/ab', () => ({
+	isInVariantSynchronous: jest.fn(),
+}));
+
+const canTargetTCFv2 = (canTarget: boolean): ConsentState => ({
+	tcfv2: {
+		consents: { 1: canTarget, 2: canTarget, 3: canTarget },
+		gdprApplies: true,
+		eventStatus: 'tcloaded',
+		vendorConsents: { abc: true },
+		tcString: 'testTcString',
+		addtlConsent: 'testaddtlConsent',
+	},
+});
+
+const canTargetCCPA = (canTarget: boolean): ConsentState => ({
+	ccpa: {
+		doNotSell: !canTarget,
+	},
+});
+
+const canTargetAUS = (canTarget: boolean): ConsentState => ({
+	aus: {
+		personalisedAdvertising: canTarget,
+	},
+});
+
+describe('create ads config', () => {
```

Comment:
Nice clear test cases explaining what's going on 👍

Reply:
Most of the credits goes to @arelra–I’ve only changed a `j` to a `t` in the extension.